fix(lemon-ui): Update lemon input select to use virtualized list

lshaowei18 opened the pull request:
> [!IMPORTANT]
> 👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Problem

Closes #34681

The timezones that are shown was limited to 100 due to a quick fix as it was getting slower after about ~10k rows. 

## Changes

Made the rendering of `visibleOptions` a virtualized list:


https://github.com/user-attachments/assets/57d1731e-dc63-4d9e-8542-ec95325c5363

This is my first time using this virtualized library, so please do point out any things that can be improved!

## Did you write or update any docs for this change?
- [x] No docs needed for this change

## How did you test this code?

Locally. I did a test where i loaded `visibleOptions` with > 10k options, it didn't slow down the performance. However if you navigate too fast with the scrollbar the items might not appear.

lshaowei18 commented on frontend/src/lib/lemon-ui/LemonInputSelect/LemonInputSelect.tsx:
```diff
@@ -472,57 +469,84 @@ export function LemonInputSelect({
+                                                                checked={isSelected}
+                                                                className="pointer-events-none"
+                                                            />
+                                                        ) : undefined
+                                                    }
+                                                    sideAction={
+                                                        !option.__isInput && allowCustomValues
+                                                            ? {
+                                                                  // To reduce visual clutter we only show the icon on focus or hover,
+                                                                  // but we do want it present to make sure the layout is stable
+                                                                  icon: (
+                                                                      <IconPencil
+                                                                          className={
+                                                                              !isFocused ? 'invisible' : undefined
+                                                                          }
+                                                                      />
+                                                                  ),
+                                                                  tooltip: (
+                                                                      <>
+                                                                          Edit this value{' '}
+                                                                          <KeyboardShortcut option enter />
+                                                                      </>
+                                                                  ),
+                                                                  onClick: () => {
+                                                                      setInputValue(option.key)
+                                                                      inputRef.current?.focus()
+                                                                      _onFocus()
+                                                                  },
+                                                              }
+                                                            : undefined
+                                                    }
+                                                >
+                                                    <span className="whitespace-nowrap ph-no-capture truncate">
+                                                        {!option.__isInput
+                                                            ? option.labelComponent ?? option.label // Regular option
+                                                            : mode === 'multiple'
+                                                            ? `Add "${option.key}"` // Input-based option
+                                                            : option.key}
+                                                    </span>
+                                                </LemonButton>
```
Hmmmm I don't think this is true; although it might be a possibility that `LemonButton` does not include the `style` prop. (I checked, it does include)

Twixes reviewed (commented):
Makes sense at a glance for me (left a height comment), but will let @PostHog/devex ultimately approve, as I'm not on the component library side of things anymore much

lshaowei18 commented on frontend/src/lib/lemon-ui/LemonInputSelect/LemonInputSelect.tsx:
```diff
@@ -472,57 +469,84 @@ export function LemonInputSelect({
-                                </LemonButton>
-                            )
-                        })
+                        <div className="min-h-[420px]">
```
Hey @Twixes, thanks for the review. Shouldn't have blindly copied the value from another component, my bad. 

You made a good point, I have now updated the component to calculate dynamically based on the number of items in the select.

<img width="1052" height="448" alt="Screenshot 2025-07-11 at 9 57 34 AM" src="https://github.com/user-attachments/assets/d49ed3f6-1541-49ee-9567-c224380b2204" />

lshaowei18 committed:
fix: add option to virtualize lemon input select and turn it on for timezone config

lshaowei18 commented:
Hi @mariusandra , I have updated the following:

1) The virtualized item's height is now set to `33`, which i believe is the same as the button size small.
2) I have added an optional prop `virtualized`

Great feedback, appreciate it; i definitely think this is way safer.

lshaowei18 committed:
fix: add comment about magic number

lshaowei18 committed:
Merge branch 'master' into lemon-input-select-virtual

lshaowei18 committed:
Merge branch 'master' into lemon-input-select-virtual

adamleithp commented:
Well done looks great

lshaowei18 committed:
Merge branch 'master' into lemon-input-select-virtual
